### PR TITLE
Feature prometheus

### PR DESCRIPTION
**Pull request by devops-dmitry**
alerts and inhibitRules have been moved to this chart. minor code errors have been fixed. prometheus has been updated.
added interface to remove default alerts

### CODE REVIEW

File: chart_deps/konghq/plugins/values.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Declare variables to be passed into your templates.
 # Configur KongPlugins CRD
 
-plugins: []
+plugins: {}
```
Comment (ggramal) [inline chart_deps/konghq/plugins/values.yaml:6]: Hmm why are we changing this attr type here :thinking: ?

File: chart_deps/prometheus/prometheus-rules/templates/prometheus-alerts.yaml [hunk 2]
```diff
@@ -65,21 +65,21 @@ spec:
       interval: {{ .WarningMetric.interval }}
       {{- end }}
       rules:
-  {{- range .WarningMetric.rules }}
-        - alert: {{.name}}
+        {{- range $key, $val := .WarningMetric.rules }}
```
Comment (ggramal) [inline chart_deps/prometheus/prometheus-rules/templates/prometheus-alerts.yaml:68]: same as https://github.com/ggramal/helm-charts/pull/7/files#r1615639168

File: prometheus/values.yaml [hunk 3]
```diff
@@ -276,5 +314,83 @@ prometheus-postgres-exporter:
...
+            expr: sum by (pod, container, namespace, reason) (kube_pod_container_status_waiting_reason{exported_namespace="kube-system", reason!="OOMKilled"})>0
+          Pod_replicas_not_ready:
+            for: "15m"
+            description: "Pods replicas less than specified for '{{ $labels.deployment }}' for > 15m"
+            expr: (kube_deployment_spec_replicas * on (deployment) group_left (container) kube_deployment_labels) != (kube_deployment_status_replicas_available * on (deployment) group_left (container) kube_deployment_labels)
+          Daemonset_not_ready:
+            for: "15m"
+            description: "daemonset '{{ $labels.daemonset }}' not scheduled in 15m"
+            expr: kube_daemonset_status_number_misscheduled > 1
+          RabbitMQ_High_memory(watermark)_usage:
```
Comment (ggramal) [inline prometheus/values.yaml:358]: What will happen if a company/project/env does not have Rabbitmq?

**Comment by ggramal** [comment]
Apart from small comments, everything looks very good. So to summarize
1. We introduce a braking change (alerts have type dict now not list)
2. We update our prometheus/thanos stack
Are there any braking changes in point 2?

File: prometheus/values.yaml [hunk 4]
```diff
@@ -206,8 +245,8 @@ prometheus-main:
     spec:
       enableAdminAPI: true
       thanos:
-        image: quay.io/thanos/thanos:v0.34.1
-        version: v0.34.1
+        image: quay.io/thanos/thanos:v0.35.0
```
Comment (ggramal) [inline prometheus/values.yaml:248]: do we need to also update thanos chart?

File: prometheus/values.yaml (hunk 3 above)
Comment (devops-dmitry) [reply prometheus/values.yaml:358]: The alert will not work

**Comment by devops-dmitry** [comment]
> Apart from small comments, everything looks very good. So to summarize
> 
> 1. We introduce a braking change (alerts have type dict now not list)
> 2. We update our prometheus/thanos stack
>    Are there any braking changes in point 2?
yes (Alertmanager 0.27.0):
```
Removal of API v1
The Alertmanager v1 API has been deprecated since January 2019 with the release of Alertmanager v0.16.0. With the release of version 0.27.0 it is now removed.
A successful HTTP request to any of the v1 endpoints will log and return a deprecation message while responding with a status code of 410.
Please ensure you switch to the v2 equivalent endpoint in your integrations before upgrading.
```

File: prometheus/values.yaml (hunk 4 above)
Comment (devops-dmitry) [reply prometheus/values.yaml:248]: Yes. I missed it

File: chart_deps/konghq/plugins/values.yaml (hunk 1 above)
Comment (devops-dmitry) [reply chart_deps/konghq/plugins/values.yaml]: return [ ]

File: chart_deps/prometheus/prometheus-rules/templates/prometheus-alerts.yaml [hunk 5]
```diff
@@ -42,21 +42,21 @@ spec:
       interval: {{ .CriticalMetric.interval }}
       {{- end }}
       rules:
-  {{- range .CriticalMetric.rules }}
-        - alert: {{.name}}
+        {{- range $key, $val := .CriticalMetric.rules }}
```
[the remark this answers is not in the document]
Comment (devops-dmitry) [reply chart_deps/prometheus/prometheus-rules/templates/prometheus-alerts.yaml]: ok. changed

File: chart_deps/prometheus/prometheus-rules/templates/prometheus-alerts.yaml (hunk 2 above)
Comment (devops-dmitry) [reply chart_deps/prometheus/prometheus-rules/templates/prometheus-alerts.yaml]: changed

File: prometheus/values.yaml (hunk 4 above)
Comment (devops-dmitry) [reply prometheus/values.yaml:248]: done

**Review by ggramal — APPROVED** [review]

**Pull request merged**